### PR TITLE
[WIP] Register cloudant-spark with Spark Packages community index

### PR DESCRIPTION
**Pull request by mayya-sharipova**
Following instructions in the
https://github.com/databricks/sbt-spark-package

ToDo:
1) Create a file / ".ivy2" / ".sbtcredentials" with Github
cloudant-service user credentials

Registering and publishing Spark packages:
1) sbt spDist - creates a zip archive ready for a release on the Spark
packages website. The zip file will be produced in <project>/target.
2) sbt spRegister - registering Spark package in to the Spark Packages
website for the FIRST time only.
3) sbt spPublish - publish a new release.  First you should have pushed
your commit to the master branch on your remote, and you must be located
on the master branch.

After that anyone can use our spark package in pyspark or spark-submit
by using --packages com.cloudant.spark/cloudant-spark:0.1-SNAPSHOT

BugzID: 53022

### CODE REVIEW

**Comment by HolgerKache** [comment]
@mayya-sharipova 
I was able to build locally and get `cloudant-spark-0.1-SNAPSHOT-s_2.10.zip` created. Assuming that the build server will have the release tag, it should become something like `cloudant-spark-1.6.1-43.zip` with the appropriate `cloudant-spark-1.6.1-43.pom` contained, right?

Can you please go ahead and merge this and also trigger a release build immediately after the merge? Call it v1.6.1.1 and make sure to create the release notes including all the changes we delivered since v1.6.1.0. That build can then go out to spark-packages.org into http://spark-packages.org/package/cloudant-labs/spark-cloudant (manual upload or via sbt spPublish)

Once we know the process works and the release on spark-packages.org is working as expected, we should look to improve our build scripts to automate the push out to spark-packages.org

File: cloudant-spark-sql/build.sbt [hunk 1]
```diff
@@ -37,3 +37,25 @@ assemblyMergeStrategy in assembly := {
...
+spDescription := """Spark SQL Cloudant External Datasource.
+                   |Cloudant integration with Spark as Spark SQL external datasource.
+                   | - Allows to load data from Cloudant dabases and indexes into Spark.
+                   | - Allows to save resulting data from Spark to existing Cloudant databases.
+                   | - Supports predicates push down (only based on _id field in databases, but varios fields for indexes).
+                   | - Support column pruning for indexes.""".stripMargin
+
+credentials += Credentials(Path.userHome / ".ivy2" / ".sbtcredentials")
+
+
```
Comment (HolgerKache) [inline cloudant-spark-sql/build.sbt:60]: I tried `sbt spPublish` but get an exception: java.lang.AssertionError: assertion failed: Please provide a license with the key licenses in your build.sbt file

Can we add `licenses += "Apache-2.0" -> url("http://opensource.org/licenses/Apache-2.0")` please?

File: cloudant-spark-sql/build.sbt [hunk 2]
```diff
@@ -37,3 +37,25 @@ assemblyMergeStrategy in assembly := {
 }
 
 assemblyJarName in assembly := "cloudant-spark.jar"
+
+
+spAppendScalaVersion := true
+
+spName := "com.cloudant.spark/cloudant-spark"
```
Comment (hkache) [inline cloudant-spark-sql/build.sbt]: I think this has to be `cloudant-labs/spark-cloudant`

File: cloudant-spark-sql/build.sbt [hunk 3]
```diff
@@ -1,6 +1,6 @@
 organization  := "com.cloudant.spark"
```
Comment (hkache) [inline cloudant-spark-sql/build.sbt]: this should be `cloudant-labs`

File: cloudant-spark-sql/build.sbt [hunk 4]
```diff
@@ -1,6 +1,6 @@
...
 
-name := "spark-sql"
+name := "cloudant-spark"
```
Comment (hkache) [inline cloudant-spark-sql/build.sbt:4]: name has to be `spark-cloudant` to comply with our package http://spark-packages.org/package/cloudant-labs/spark-cloudant

File: cloudant-spark-sql/build.sbt (hunk 1 above)
Comment (mayya-sharipova) [inline cloudant-spark-sql/build.sbt:60]: thanks, corrected

File: cloudant-spark-sql/build.sbt (hunk 2 above)
Comment (mayya-sharipova) [inline cloudant-spark-sql/build.sbt]: thank you for the comment, corrected

File: cloudant-spark-sql/build.sbt (hunk 3 above)
Comment (mayya-sharipova) [inline cloudant-spark-sql/build.sbt]: thank you for the comment, corrected

File: cloudant-spark-sql/build.sbt (hunk 4 above)
Comment (mayya-sharipova) [inline cloudant-spark-sql/build.sbt:4]: thank you for the comment, corrected

**Pull request merged**